crates/ra_hir/src/ty/lower.rs

```diff
@@ -185,6 +163,80 @@ impl Ty {
     }
 }
 
+pub(super) fn substs_from_path_segment(
+    db: &impl HirDatabase,
+    resolver: &Resolver,
+    segment: &PathSegment,
+    def_generics: &GenericParams,
+    add_self_param: bool,
+) -> Substs {
+    let mut substs = Vec::new();
+    let parent_param_count = def_generics.count_parent_params();
+    substs.extend((0..parent_param_count).map(|_| Ty::Unknown));
```

Comment:
Perhaps `iter::repeat(Ty::Unknwn).take(parent_param_count)` would be more direct